lib/versioning/regex/index.ts

```diff
@@ -66,66 +66,95 @@ export class RegExpVersioningApi extends GenericVersioningApi<RegExpVersion> {
 
   // convenience method for passing a string into a Version given current config.
   protected _parse(version: string): RegExpVersion | null {
-    const match = this._config.exec(version);
-    if (match === null) {
+    const groups = this._config?.exec(version)?.groups;
+    if (!groups) {
       return null;
     }
 
-    const groups = match.groups;
+    const { major, minor, patch, build, prerelease, compatibility } = groups;
     const release = [
-      typeof groups.major === 'undefined' ? 0 : Number(groups.major),
-      typeof groups.minor === 'undefined' ? 0 : Number(groups.minor),
-      typeof groups.patch === 'undefined' ? 0 : Number(groups.patch),
+      typeof major === 'undefined' ? 0 : Number(major),
+      typeof minor === 'undefined' ? 0 : Number(minor),
+      typeof patch === 'undefined' ? 0 : Number(patch),
     ];
 
-    if (groups.build) {
-      release.push(Number(groups.build));
+    if (build) {
+      release.push(Number(build));
     }
 
     return {
       release,
-      prerelease: groups.prerelease,
-      compatibility: groups.compatibility,
+      prerelease: prerelease,
+      compatibility: compatibility,
     };
   }
 
   override isCompatible(version: string, current: string): boolean {
-    return (
-      this._parse(version).compatibility === this._parse(current).compatibility
+    const parsedVersion = this._parse(version);
+    const parsedCurrent = this._parse(current);
+    return !!(
+      parsedVersion &&
+      parsedCurrent &&
+      parsedVersion.compatibility === parsedCurrent.compatibility
     );
   }
 
   override isLessThanRange(version: string, range: string): boolean {
-    return semver.ltr(
-      asSemver(this._parse(version)),
-      asSemver(this._parse(range))
+    const parsedVersion = this._parse(version);
+    const parsedRange = this._parse(range);
+    return !!(
+      parsedVersion &&
+      parsedRange &&
+      semver.ltr(asSemver(parsedVersion), asSemver(parsedRange))
     );
   }
 
+  private getSemverVersions(versions: string[]): string[] {
```

Comment:
```suggestion
  private _getSemverVersions(versions: string[]): string[] {
```

i think we skip functions with underscore in tests